cmd/krew/cmd/internal/downloadFile.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2019 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"io"
+	"net/http"
+)
+
+// DownloadFile function takes URL as input, stores the contents in temporary file and returns filename, cleanup function to delete temporary files and error
+func DownloadFile(url string) (io.ReadCloser, error) {
+	resp, err := http.Get(url)
```

Comment:
`http.Get` returns a response object whose `Body` is expected to be closed by the user. See the documentation for `http.Do` for details:

> // If the returned error is nil, the Response will contain a non-nil
// Body which the user is expected to close. If the Body is not both
// read to EOF and closed, the Client's underlying RoundTripper
// (typically Transport) may not be able to re-use a persistent TCP
// connection to the server for a subsequent "keep-alive" request.

I think it would make more sense to not just fetch any file with this method, but instead move the whole plugin parsing (`indexscanner.DecodePluginFile`) into this function (see the previous comment). This also improves the testability.

Consequently, the method should be called `ReadRemotePluginManifest` or similar.

Reply:
Done